src/components/layout/UploadThumb/__specs__/UploadThumb.spec.js

```diff
@@ -0,0 +1,396 @@
+  let props
+  let image = {
+    lastModified: 1565378546222,
+    lastModifiedDate: 'Fri Aug 09 2019 21:22:26 GMT+0200 (heure d’été d’Europe centrale)',
+    name: 'IMG_4366.jpg',
+    size: 1503804,
+    type: 'image/jpeg',
+    webkitRelativePath: '',
+  }
+
+  beforeEach(() => {
+    props = {
+      collectionName: 'fake',
+      dispatch: jest.fn(),
+      hasExistingImage: false,
+      image,
+      onImageChange: jest.fn(),
+      storeKey: 'image',
+    }
+  })
+
+  it('should match the snapshot', () => {
+    // given
+    props.image = {
+      name: 'IMG_4366.jpg',
+      size: 1503804,
+      type: 'image/jpeg',
+    }
+
+    // when
+    const wrapper = shallow(<UploadThumb {...props} />)
+
+    // then
+    expect(wrapper).toMatchSnapshot()
+  })
+
+  describe('render', () => {
+    it('should render main component with default state', () => {
+      // when
+      const wrapper = shallow(<UploadThumb {...props} />)
+
+      // then
+      expect(wrapper.state('readOnly')).toBe(false)
+      expect(wrapper.state('image')).toStrictEqual(image)
+      expect(wrapper.state('isUploadDisabled')).toBe(false)
+      expect(wrapper.state('zoom')).toBe(1)
+    })
+
+    describe('editor-zone div', () => {
+      it('should render main component properly', () => {
+        // given
+        const wrapper = shallow(<UploadThumb {...props} />)
+
+        // when
+        const editorZoneComponent = wrapper.find('.editor-zone')
+
+        // then
+        expect(editorZoneComponent).toHaveLength(1)
+      })
+
+      it('should not have no-drag class when is readOnly', () => {
+        // given
+        props.image = {
+          name: 'editor-zoneExample.jpg',
+          size: 1503804,
+          type: 'image/jpeg',
+        }
+        props.hasExistingImage = true
+
+        // when
+        const wrapper = shallow(<UploadThumb {...props} />)
+        const editorZoneComponent = wrapper.find('.editor-zone')
+
+        // then
+        expect(editorZoneComponent.props().className).toStrictEqual('editor-zone has-image no-drag')
+      })
+    })
+
+    describe('image too large alert message', () => {
+      it('should not be displayed when image is smaller', () => {
+        // when
+        const wrapper = shallow(<UploadThumb {...props} />)
+        const showAlertComponent = wrapper.find('.has-text-danger')
+
+        // then
+        expect(showAlertComponent).toHaveLength(0)
+      })
+
+      it('should not be displayed when image is larger', () => {
+        // given
+        props.image = {
+          name: 'editor-zoneExample.jpg',
+          size: 150380467,
+          type: 'image/jpeg',
+        }
+        props.hasExistingImage = true
+
+        // when
+        const wrapper = shallow(<UploadThumb {...props} />)
+        const showAlertComponent = wrapper.find('.has-text-danger')
+
+        // then
+        expect(showAlertComponent.text()).toStrictEqual(
+          'Votre image trop volumineuse : 150.380467 > 10 Mo'
```

Comment:
```suggestion
          'Votre image trop volumineuse : 150.380467 Mo > 10 Mo'
```

Reply:
C'est un détail, mais je trouve que `150.380467` n'est pas très lisible. Il faudrait l'arrondir.
Ou (plus simple) remplacer le message d'erreur par : 
```suggestion
          'Votre image trop volumineuse, elle doit faire moins de 10 Mo.'
```

Reply:
En vrai, je pense que ça devrait être un ticket et une fonctionnalité, c'est un artefact archéologique qui revient à la surface.